### ISSUE TITLE
[Bug]: Dialog doesn't focus on list

### ISSUE THREAD

**Issue by mbest** [microsoft/fluentui#23925]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22622
    CPU: (12) x64 Intel(R) Xeon(R) CPU E5-1650 v4 @ 3.60GHz
    Memory: 9.79 GB / 31.92 GB
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Spartan (44.22621.290.0), Chromium (103.0.1264.37), ChromiumDev (104.0.1293.5)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/mbest098/pen/WNzozwO

## Actual Behavior
Initial focus in dialog goes after the list

## Expected Behavior
A focusable list should receive focus if it's the first item in the dialog

### Requested priority

High

### Products/sites affected

https://account.microsoft.com/family/

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
Thanks for filing this issue with us. To set expectations the developers will review this issue once capacity allows. Thank you for showing the willingness to contribute to this request. The developers will review the issue and if required will work with you on submitting a PR for the fix.

**Comment by layershifter** [comment]
This problem is not with `Dialog` itself - it comes from a pair of `List` & `FocusZone`, here is a minimal repro: https://codesandbox.io/s/pensive-hugle-buxqdx.

### What happens? 🕵️‍♂️

- `List` component renders without `pages` (they are computed on second render after `componentDidUpdate()`
  ![image](https://user-images.githubusercontent.com/14183168/184880549-38517d75-00aa-4bfe-8abc-330d0a621a93.png)
- `FocusZone` calls `_updateTabIndexes()` to assign `tabIndex`es, but it assigns nothing as there no child elements
- `List` renders again
- nothing happens, elements in `List` cannot be focused

### How to repro? 💡
- Open https://codesandbox.io/s/pensive-hugle-buxqdx
- Inspect elements inside `List`, there are no `tabIndex`es, nothing can be focused by keyboard 🔴
- Click on a random element in `List`, `FocusZone` will call `componentDidUpdate` and assign `tabIndex`es
- Now elements can be focused by keyboard and navigation works

----

I searched through issues and code, but failed to find a solution. All examples in docs use `FocusZone` in the same 🤔 

I will reach out to folks internally to get some suggestions around it.

**Comment by layershifter** [comment]
### Update 🆕 

Focus behavior change in the previous CodeSandbox is a regression caused by #24175.

- 8.66.3 https://codesandbox.io/s/intelligent-chebyshev-mjpq47
- 8.86.2 https://codesandbox.io/s/hardcore-tu-x5dv5n

However, the problem from the previous message with `Dialog` is present even in 8.86.2, https://codesandbox.io/s/frosty-marco-0yvwdq

**Comment by micahgodbolt** [comment]
@mbest Sean was able to work up a decent workaround by waiting for the pages to load and then calling the FocusZone's focus method.

https://codepen.io/seanms/pen/dymEVWR

If this solution works, we could consider creating a callback that basically does all this for you, though this workaround should work perfectly well for now.

**Comment by mbest** [comment]
@micahgodbolt That is basically the workaround that I used in our app, but perhaps we could consider another fix, such as having `List` output content earlier.

**Comment by spmonahan** [comment]
@mbest I've been looking at this recently and have a potential solution that I'd like to hear from you about. Can you take a look at #25331 ?

**Comment by spmonahan** [comment]
Related: #26899
